onnxruntime/core/optimizer/attention_fusion.cc

```diff
@@ -413,16 +413,14 @@ bool AttentionFusion::FuseSubGraph(Node& layer_norm, const Node& add_after_layer
     return false;
   }
 
-  if (add.GetOutputEdgesCount() != 1 ||
-      matmul.GetOutputEdgesCount() != 1 ||
-      reshape.GetOutputEdgesCount() != 1 ||
-      transpose.GetOutputEdgesCount() != 1 ||
-      qkv_matmul.GetOutputEdgesCount() != 1 ||
-      v_transpose.GetOutputEdgesCount() != 1 ||
-      v_reshape.GetOutputEdgesCount() != 1 ||
-      v_add.GetOutputEdgesCount() != 1 ||
-      v_matmul.GetOutputEdgesCount() != 1 ||
-      v_root.GetOutputEdgesCount() != 4) {
+  // Internal nodes of attention subgraph only allow edges within the subgraph, and no graph output is allowed.
```

Comment:
should these nodes be exempted from the number of output edges check as well? .i.e. add.GetOutputEdgesCount() != 1

Reply:
There are comments that the constraints are removed for outside nodes (add, matmul and v_root) not to be fused.